servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -422,27 +429,29 @@ public EntriesResponse getEntries(String namedRef, EntriesParams params)
     // to the store though
     //  all existing VersionStore implementations have to read all keys anyways so we don't get much
     try {
-      List<EntriesResponse.Entry> entries;
-      try (Stream<EntriesResponse.Entry> entryStream =
-          getStore()
-              .getKeys(refWithHash.getHash())
-              .map(
-                  key ->
-                      EntriesResponse.Entry.builder()
-                          .name(fromKey(key.getValue()))
-                          .type((Type) key.getType())
-                          .build())) {
-        Stream<EntriesResponse.Entry> entriesStream = filterEntries(entryStream, params.filter());
+      ImmutableEntriesResponse.Builder response = EntriesResponse.builder();
+      try (Stream<KeysEntry<Content.Type>> entryStream =
+          getStore().getKeys(refWithHash.getHash())) {
+        Stream<KeysEntry<Content.Type>> keysEntryStream =
+            filterEntries(refWithHash, entryStream, params.filter());
+
+        Stream<EntriesResponse.Entry> entriesStream =
+            keysEntryStream.map(
```

Comment:
could we just combine `keysEntryStream`  with `entriesStream`?
`Stream<EntriesResponse.Entry> entriesStream = filterEntries(refWithHash, entryStream, params.filter()).map(...)`

Reply:
yup